From ea28ffadc7f9fea9f3c68ae2448128454f8d7341 Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Wed, 22 Nov 2017 15:22:12 +0800
Subject: [PATCH 2875/5242] MLK-16942-1: hdmi-cec: Update clock rate to 133MHz

commit  4996198715ceafb7723759e0d2eb1133260a79b2 from
https://source.codeaurora.org/external/imx/linux-imx.git

After apply HDMI FW ROM patch to mscale B0.
The HDMI core clock will run at 133MHz.
Update hdmi cec clock rate.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
(cherry picked from commit 4198c8f6677eff3e6e50ca17f3d1b7f6e6f8b8c3)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/hdp/imx-cec.c |    7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/imx/hdp/imx-cec.c b/drivers/gpu/drm/imx/hdp/imx-cec.c
index 95b1efa..9d8c1592 100644
--- a/drivers/gpu/drm/imx/hdp/imx-cec.c
+++ b/drivers/gpu/drm/imx/hdp/imx-cec.c
@@ -285,11 +285,8 @@ static int imx_cec_probe(struct platform_device *pdev)
 	if (IS_ERR(cec->reg_base))
 		return PTR_ERR(cec->reg_base);
 
-	/* hdmi core clock is 125MHz in B0 */
-	if (imx8_get_soc_revision() == B0_SILICON_ID)
-		cec->clk_div = 1250;
-	else
-		cec->clk_div = 1330;
+	/* hdmi core clock is 133MHz */
+	cec->clk_div = 1330;
 
 	cec->adap = cec_allocate_adapter(&imx_cec_adap_ops, cec,
 					 CEC_NAME,
-- 
1.7.9.5

